starlette/exceptions.py

```diff
@@ -81,15 +81,16 @@ def _lookup_exception_handler(
             if handler is None:
                 raise exc from None
 
-            if response_started:
-                msg = "Caught handled exception, but response already started."
-                raise RuntimeError(msg) from exc
-
             request = Request(scope, receive=receive)
             if asyncio.iscoroutinefunction(handler):
                 response = await handler(request, exc)
             else:
                 response = await run_in_threadpool(handler, request, exc)
+
+            if response_started:
```

Comment:
Not clear to me that we should actually even raise this error here, anymore?

Reply:
I'm going to suggest that we drop it. We're calling the error handler - that's sufficient.

Reply:
So instead, just...

```python
if not response_started:
    await response(scope, receive, sender)
```